controllers/templatesync/template_sync.go

```diff
@@ -1507,3 +1526,27 @@ func finalizerCleanup(
 func (r *PolicyReconciler) setCreatedGkConstraint(b bool) {
 	r.createdGkConstraint = &b
 }
+
+// Check duplicate names in policy-templates(configurationPolicies)
+func hasDupName(pol *policiesv1.Policy) (bool, error) {
+	templates := pol.Spec.PolicyTemplates
+
+	empty := make(map[string]struct{})
```

Comment:
Nit: naming this `empty` is a little confusing. My preference would be `foundNames`.